tests/backup/backup_test.go

```diff
@@ -6514,6 +6514,175 @@ var _ = Describe("{IssueMultipleDeletesForSharedBackup}", func() {
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share the backup with same name", func() {
+		providers := getProviders()
+		buckets := getBucketName()
+		Step("Create Users", func() {
+			log.InfoD("Creating %d users", numberOfUsers)
+			var wg sync.WaitGroup
+			for i := 1; i <= numberOfUsers; i++ {
+				userName := fmt.Sprintf("testuser%v", i)
+				firstName := fmt.Sprintf("FirstName%v", i)
+				lastName := fmt.Sprintf("LastName%v", i)
+				email := fmt.Sprintf("testuser%v@cnbu.com", i)
+				wg.Add(1)
+				go func(userName, firstName, lastName, email string) {
+					defer wg.Done()
+					err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+					log.FailOnError(err, "Failed to create user - %s", userName)
+					users = append(users, userName)
+				}(userName, firstName, lastName, email)
+			}
+			wg.Wait()
+		})
+		Step(fmt.Sprintf("Adding Credentials and Registering Backup Location for %s and %s", users[0], users[1]), func() {
+			log.InfoD(fmt.Sprintf("Creating bucket, cloud credentials and backup location for %s and %s", users[0], users[1]))
+			for _, provider := range providers {
+				BucketName = fmt.Sprintf("%s-%v", provider, buckets[0])
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+
+				for _, user := range users {
+					credName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+					backup.AddRoleToUser(user, backup.InfrastructureOwner, fmt.Sprintf("Adding Infra Owner role to %s", user))
+					NonAdminUserctx, err := backup.GetNonAdminCtx(user, "Password1")
+					log.FailOnError(err, fmt.Sprintf("Failed to fetch ctx for custom user: [%v]", err))
+					CreateCloudCredentialNonAdminUser(provider, credName, cloudCredUID, orgID, NonAdminUserctx)
+					credNames = append(credNames, credName)
+					time.Sleep(time.Minute * 1)
+					backupLocationName := fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+					CreateS3BackupLocationNonAdminUser(backupLocationName, backupLocationUID, credName, cloudCredUID, BucketName, orgID, "", NonAdminUserctx)
+					userbackupLocationMapping[user] = backupLocationName
+				}
+
+			}
+		})
+		for _, user := range users {
+			Step(fmt.Sprintf("Register source and destination cluster for backup on %s ", user), func() {
+				log.InfoD("Registering Source and Destination clusters as user : %s and verifying the status", user)
+				ctx, err := backup.GetNonAdminCtx(user, "Password1")
+				log.FailOnError(err, "Fetching %s ctx", user)
+				CreateSourceAndDestClusters(orgID, "", "", ctx)
+				time.Sleep(time.Minute * 2)
+				clusterStatus, clusterUid = Inst().Backup.RegisterBackupClusterNonAdminUser(orgID, SourceClusterName, "", ctx)
+				dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+			})
+			Step(fmt.Sprintf("Taking backup of applications as %s", user), func() {
+				log.InfoD("Taking backup of applications as user : %s ", user)
+				backupName = "backup1"
```

Comment:
I would suggest giving a backup name which is unique and easily recognisable. Like `backup-<testrailID>`